pkg/destroy/ibmcloud/dns.go

```diff
@@ -0,0 +1,97 @@
+package ibmcloud
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const dnsRecordTypeName = "dns record"
+
+// listDNSRecords lists DNS records for the cluster
+func (o *ClusterUninstaller) listDNSRecords() (cloudResources, error) {
+	o.Logger.Debugf("Listing DNS records")
+	ctx, cancel := o.contextWithTimeout()
+	defer cancel()
+
+	result := []cloudResource{}
+	moreData := true
+	for moreData {
+		options := o.dnsRecordsSvc.NewListAllDnsRecordsOptions()
+		resources, _, err := o.dnsRecordsSvc.ListAllDnsRecordsWithContext(ctx, options)
+
+		if err != nil {
+			return nil, errors.Wrapf(err, "Failed to list DNS records")
+		}
+
+		for _, record := range resources.Result {
+			if strings.Contains(*record.Name, o.ClusterName) ||
+				strings.Contains(*record.Content, o.ClusterName) {
```

Comment:
The cluster ID is a GUID (not in the record name or content. What do you mean by cluster ID?

Reply:
This is the metadata we have access to for deleting resources:
```json
{
  "clusterName": "ocp4-ipi-072103",
  "clusterID": "5ee7b41c-1699-4b79-8530-06775548e232",
  "infraID": "ocp4-ipi-072103-b6576",
  "ibmcloud": {
    "accountID": "1e1f75646aef447814a6d907cc83fb3c",
    "baseDomain": "ibm.bobbyradford.com",
    "cisInstanceCRN": "crn:v1:bluemix:public:internet-svcs:global:a/1e1f75646aef447814a6d907cc83fb3c:5a8989d1-54c2-4634-9a33-63c57c5f129a::",
    "region": "us-east",
    "resourceGroupName": "ocp4-ipi-072103-b6576"
  }
}
```

If you mean `infraID`, that won't work because the DNS record uses the `cluster_name` TF var, which doesn't include the random suffix that the infraID has:

```json
{
"cluster_domain": "ocp4-ipi-072103.ibm.bobbyradford.com"
}
```

Reply:
> The cluster ID is a GUID (not in the record name or content. What do you mean by cluster ID?

Ah I see, yes I was confused with the naming there. I was referring to the `infraID`.

Reply:
> If you mean infraID, that won't work because the DNS record uses the cluster_name TF var, which doesn't include the random suffix that the infraID has...

Sorry, which part will not work? I think it should look for DNS records with..
- {infraID}-master-{index}.{clusterName}
- api.{clusterName}
- api-int.{clusterName}

..or we can simply delete all DNS records under the cluster subdomain. For this, we need to make sure the `clusterName` string value is located in the proper segment in the subdomain.

e.g. (if clusterName = "cluster1")
- api.cluster1 (Match)
- api.cluster1.test (No match)

Reply:
OK. I see what you're saying. I'm onboard.

Reply:
Added logic to support your second suggestion here.